src/data/token.tsx

```diff
@@ -73,12 +74,22 @@ export const useNativeDenoms = () => {
   const { whitelist, ibcDenoms, legacyWhitelist } = useWhitelist()
   const { list: cw20 } = useCustomTokensCW20()
   const networkName = useNetworkName()
+  const networks = useNetwork()
 
   function readNativeDenom(denom: Denom): TokenItem {
     const fixedDenom = denom.startsWith("ibc/")
       ? `${readDenom(denom).substring(0, 5)}...`
+      : denom.startsWith("factory/")
+      ? denom.split("/").pop()?.slice(1).toUpperCase()
```

Comment:
This is the right idea.  Unfortunately there is no standard for what happens after "factory".

For example, on osmosis, there are several factory tokens for compounding vaults.  Their denom string is:
`factory/osmo1g3kmqpp8608szfp0pdag3r6z85npph7wmccat8lgl3mp407kv73qlj7qwp/VaultToken/1/14d/ATOM/OSMO`

While this conforms to its own standard, it illustrates that almost anything can come after `factory/`